vault/activity/query.go

```diff
@@ -134,6 +134,30 @@ func (s *PrecomputedQueryStore) listEndTimes(ctx context.Context, startTime time
 	return endTimes, nil
 }
 
+func (s *PrecomputedQueryStore) getMaxEndTime(ctx context.Context, startTime time.Time, endTimeBound time.Time) (time.Time, error) {
+	rawEndTimes, err := s.view.List(ctx, fmt.Sprintf("%v/", startTime.Unix()))
+	if err != nil {
+		return time.Time{}, err
+	}
+
+	maxEndTime := time.Time{}
+	for _, raw := range rawEndTimes {
+		val, err := strconv.ParseInt(raw, 10, 64)
+		if err != nil {
+			s.logger.Warn("could not parse precomputed query end time", "key", raw)
+			continue
+		}
+		endTime := time.Unix(val, 0).UTC()
+		s.logger.Trace("end time in consideration is", "end time", endTime, "end time bound", endTimeBound)
+		if endTime.After(maxEndTime) && (endTime.Before(endTimeBound) || endTime.Equal(endTimeBound)) {
```

Comment:
We can probably simplify this check to `endTime.After(maxEndTime) && !endTime.After(endTimeBound)`. It might be a bit clearer to read and removes the potential need for 3 comparisons. What do you think?

Reply:
Sure, sounds good!